raft/leaderElection/leaderelection_test.go

```diff
@@ -0,0 +1,45 @@
+package leaderelection
+
+import (
+	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
+	"testing"
+	"time"
+)
+
+func TestIncrementTermAndBecomeCandidate(t *testing.T) {
+}
+
+func TestGetTimer(t *testing.T) {
+	minTimeout := 150
+	maxTimeout := 650
+
+	// Test if timer is within expected bounds
+	timer := getTimer(minTimeout, maxTimeout)
+	timerDuration := timer.getDuration()
+
+	if timerDuration < minTimeout*Time.Millsecond {
+		t.Error("Timer runs for less time than expected")
+	} else if timerDuration > maxTimeout*Time.Millisecon {
+		t.Error("Timer runs for more time than expected")
+	}
+
+	// Test if timer is random
+}
+
+func TestFindLeader(t *testing.T) {
+	// How do I simulate votes?
```

Comment:
I need to simulate votes here. 

How exactly do I proceed because the function internally calls requestVotes which needs http responses. Any ideas?

Most of the findLeader rather belongs to LeaderElection function and the scope of this should be to just count votes from a vote channel.

Reply:
`TestFindLeader` must basically check whether a leader is elected across the cluster.
Repeating the actual process in the implementation seems quite fine here. 
But we probably can test it for multiple cluster sizes, even with the ones where the leader mustn't be elected (say simulating a split vote - or - where there aren't enough alive servers in the clusters)

Make sure to include all the above cases to be thorough.

Reply:
Problem solves itself neatly when I move code back to LeaderElection.